src/main/java/com/cloudbees/jenkins/plugins/BitBucketPayload.java

```diff
@@ -0,0 +1,37 @@
+package com.cloudbees.jenkins.plugins;
+
+import hudson.EnvVars;
+import hudson.model.AbstractBuild;
+import hudson.model.EnvironmentContributingAction;
+import hudson.model.InvisibleAction;
+
+import javax.annotation.Nonnull;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Inject the payload received by BitBucket into the build through $BITBUCKET_PAYLOAD so it can be processed
+ * @since January 9, 2016
+ * @version 1.1.5
+ */
+public class BitBucketPayload extends InvisibleAction implements EnvironmentContributingAction {
+    private final @Nonnull String payload;
+
+    public BitBucketPayload(@Nonnull String payload) {
+        this.payload = payload;
+    }
+
+    @Nonnull
+    public String getPayload() {
+        return payload;
+    }
+
+    @Override
+    public void buildEnvVars(AbstractBuild<?, ?> abstractBuild, EnvVars envVars) {
```

Comment:
In workflow you can use `currentBuild.rawBuild.getAction(BitBucketPayload).payload` to retrieve the value.